File: examples/cesar.java
import java.util.regex.Pattern;

public static String cypher(String input, int offset, String charset) {
    StringBuilder str = new StringBuilder();
    for(int i = 0; i < input.length(); i++) {
        if(Pattern.compile("\\s").matcher(Character.toString(input.charAt(i))).matches()) {
            str.append(input.charAt(i));
            continue;
        }

        int j = charset.indexOf(input.charAt(i));
        str.append(charset.charAt(
            (j + offset + charset.length()) % charset.length()
        ));
    }
    return str.toString();
}

File: examples/cesar.kt
import java.util.regex.Pattern

fun cesar(input: String, offset: Int, charset: String): String {
    val str = StringBuilder()
    for(c in input) {
        if(Pattern.compile("\\s").matcher(Character.toString(c)).matches()) {
            str.append(c)
        }
        val j = charset.indexOf(c)
        str.append(
            charset[(j + offset + charset.length) % charset.length]
        )
    }
    return str.toString()
}


